### ISSUE TITLE
Running shiny server with R in "interactive" mode

### ISSUE THREAD

**Issue by michaelwsherman** [rstudio/shiny-server#93]
Hi all, 

A specific package I'm using requires R to be in interactive mode, even though interactivity is not necessary for the features I'm using.

This issue--of interactivity being required when it is not necessarily needed--is a common enough issue that R provides a command line argument to override the default interactivity assumption (see [the R documentation on `interactive`](http://stat.ethz.ch/R-manual/R-patched/library/base/html/interactive.html)). However, it seems there is no way for shiny server to perform a similar override. I've looked through the shiny server documentation, and I see nothing about forcing shiny server to start R in interactive mode.

Is there a way to do this with shiny server? Or is it a feature yet to be added?

Thanks!

**Comment by trestletech** [comment]
Hi Michael,

That's a great point and, unfortunately, one that's not yet baked into Shiny Server. In Shiny Server Pro, I believe this could be accomplished by adding `--interactive` to the `r_path` argument (docs [here](http://rstudio.github.io/shiny-server/latest/#r_path)) though I haven't validated that myself yet. 

We'll add this to the list.

**Comment by michaelwsherman** [comment]
Thanks @trestletech. I hope it gets added soon. No Shiny Server Pro yet, unfortunately, and my org probably wont buy it until we know it can do what we need :). Thank you!

**Comment by wch** [comment]
Have you considered asking the package author to not require R to be in interactive mode? To me that makes more sense than getting Shiny server to pretend R is interactive when it really isn't. There could be unexpected consequences of doing this; for example, the `runApp` function calls `interactive()` to decide whether to launch a browser.

**Comment by michaelwsherman** [comment]
Yes @wch I have, and that is absolutely the right way to solve the problem I'm having. But there is still the issue of Shiny development and Shiny server behaving differently. Maybe the best fix for this is for Shiny apps launched from an R session pretend to be in non-interactive mode to match the behavior the app will have on Shiny server or shinyapps.io (rather than give the option for server to launch interatively)? Not sure. Thanks!

**Comment by professorbeautiful** [comment]
You could try something like this as a *temporary* fix:
  originalInteractive = function() .Primitive("interactive")   ### from the base.
  myInteractive = function()TRUE 
          ## Or better, check sys.calls(), if called from theSpecificPackage, return TRUE, else call the native interactive().
          ## The cautions by @wch are spot-on.
  assignInNamespace("interactive", myInteractive, ns = "base")
